### DISCUSSION TITLE
How do workspace extensions work when pushing to cloud with CLI?

### DISCUSSION THREAD

**Discussion by CraigLager** [structurizr/cli#151] [Q&A] [answered]
I'm looking to try out the following setup:

```
root
    - subsystem1
        - workspace.dsl
    - subsystem2
        - workspace.dsl
   enterprise.dsl
``` 

`subsystem1/workspace.dsl` and `subsystem2/workspace.dsl` both extend the workspace in `enterprise.dsl` - the same as how the [System Catalogue example](https://github.com/structurizr/examples/tree/main/dsl/system-catalog) works.

I believe I can make this render properly in Structurizr Cloud by pushing all of the files using the CLI - however what I'm struggling to see any documentation on is:

1. Do filenames need to be unique? Can I continue to use workspace.dsl in all sub directories?
2. How would I delete an extension?

**Edit**
It seems that when I publish the files, only the last to be published file counts as the workspace - even though it's intended to be an extension.

Workspace id: 87895

**Comment by simonbrowndotje** [comment, marked as answer]
Filenames don't need to be unique and I imagine you're overwriting the cloud service workspace each time you use the CLI to push. You will need 3 separate workspaces on the cloud service; for example:

```
./structurizr.sh push -workspace subsystem1/workspace.dsl -id 123 ...
./structurizr.sh push -workspace subsystem2/workspace.dsl -id 456 ...
./structurizr.sh push -workspace enterprise.dsl -id 789 ...
```
